### PR TITLE
Adding swagger v2 model serializers and tests

### PR DESCRIPTION
**Pull request by gwendolyngoetz**
### Description

Adding serializer classes for their related models and tests.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Created unit and/or browser tests which fail without the change (if possible)

### Issue(s) this completes

Related to https://github.com/civiform/civiform/issues/5358

### CODE REVIEW

File: server/test/services/openapi/v2/serializers/ParameterSerializerTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+public class ParameterSerializerTest extends OpenApiSerializationAsserter {
+  @Test
+  public void canSerializeFullObject() throws JsonProcessingException {
+    Parameter model =
+        Parameter.builder("name1", In.QUERY, DefinitionType.STRING)
+            .markAsRequired()
+            .setFormat(Format.INT32)
+            .setDescription("lorem ipsum")
+            .build();
+    ;
```
Comment (avaleske) [inline server/test/services/openapi/v2/serializers/ParameterSerializerTest.java:19]: Stray `;`? Here and line 36

**Review by avaleske — APPROVED** [review]

File: server/test/services/openapi/v2/serializers/ParameterSerializerTest.java (hunk 1 above)
Comment (gwendolyngoetz) [reply server/test/services/openapi/v2/serializers/ParameterSerializerTest.java:19]: Yep

File: server/app/services/openapi/v2/serializers/ContactSerializer.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    super(t);
+  }
+
+  @Override
+  public void serialize(Contact value, JsonGenerator gen, SerializerProvider provider)
+      throws IOException {
+    // open root
+    gen.writeStartObject();
+
+    writeStringFieldIfPresent(gen, "name", value.getName());
```
[the remark this answers is not in the document]
Comment (gwendolyngoetz) [reply server/app/services/openapi/v2/serializers/ContactSerializer.java:24]: None of the classes serialize with annotations. Often there are free-floating values based on name not the property of the class. For example some place uses http verb as a key that isn't always there or even a name.

In this case this one is much simpler especially after remove a child object we didn't need.

But I'd rather them all serialize the same way to be consistent with each other

File: server/test/services/openapi/v2/serializers/OpenApiSerializationAsserter.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      for (int i = 0; i < message.length(); i++) {
+        char c = message.charAt(i);
+        if (c == ' ') {
+          count++;
+        } else {
+          break;
+        }
+      }
+
+      return count % 2 == 0;
```
[the remark this answers is not in the document]
Comment (gwendolyngoetz) [reply server/test/services/openapi/v2/serializers/OpenApiSerializationAsserter.java]: Added it. We'll see if the tests still pass

File: server/test/services/openapi/v2/serializers/InfoSerializerTest.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            .appendLine("version: version1")
+            .appendLine("description: lorem ipsum")
+            .appendLine("contact: {}")
+            .toString();
+
+    assertSerialization(model, expected);
+  }
+
+  @Test
+  public void canSerializeEmptyObject() throws JsonProcessingException {
```
[the remark this answers is not in the document]
Comment (gwendolyngoetz) [reply server/test/services/openapi/v2/serializers/InfoSerializerTest.java:29]: Oh >_< yeah. It started out as empty, but I guess I forgot to change the name as I copied over the first as a template.

File: server/test/services/openapi/v2/serializers/OpenApiSerializationAsserter.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (gwendolyngoetz) [reply server/test/services/openapi/v2/serializers/OpenApiSerializationAsserter.java]: Success

**Pull request merged**